File: examples/run_monitor.py
# run_monitor.py
# Runs the Tucuman Compras Monitor actor and prints delivered tender records.
import os
from apify_client import ApifyClient

# Authenticate using a token from the environment (never hardcode it).
client = ApifyClient(os.environ["APIFY_TOKEN"])

# Realistic minimal input: delta mode over upcoming + in-adjudication tenders.
run_input = {
    "estados": ["1", "2"],
    "maxItems": 50,
    "onlyNew": True,
    "eventTypes": ["NEW_LISTING", "STATUS_CHANGE", "UPDATED"],
}

print("Starting tucuman-compras-monitor run...")
# Actor ID TdJtze8dfyykMj2qA — call() starts the run and waits for it to finish.
run = client.actor("TdJtze8dfyykMj2qA").call(run_input=run_input)

print(f"Run finished with status: {run['status']}")

# Pull the resulting tender records from the run's default dataset.
dataset_items = client.dataset(run["defaultDatasetId"]).list_items().items

print(f"Fetched {len(dataset_items)} tender record(s):")
for item in dataset_items:
    print(
        f"- [{item['event_type']}] {item['idCompra']} | "
        f"{item['reparticion']} | {item['tipoCompra']} | "
        f"apertura: {item['fechaAperturaSobres']}"
    )

File: examples/run-monitor.js
// run-monitor.js
// Runs the Tucuman Compras Monitor actor and logs delivered tender records.
const { ApifyClient } = require('apify-client');

// Authenticate using a token from the environment (never hardcode it).
const client = new ApifyClient({
    token: process.env.APIFY_TOKEN,
});

async function main() {
    // Realistic minimal input: delta mode over upcoming + in-adjudication tenders.
    const input = {
        estados: ['1', '2'],
        maxItems: 50,
        onlyNew: true,
        eventTypes: ['NEW_LISTING', 'STATUS_CHANGE', 'UPDATED'],
    };

    console.log('Starting tucuman-compras-monitor run...');
    // Actor ID TdJtze8dfyykMj2qA — call() starts the run and waits for it to finish.
    const run = await client.actor('TdJtze8dfyykMj2qA').call(input);

    console.log(`Run finished with status: ${run.status}`);

    // Pull the resulting tender records from the run's default dataset.
    const { items } = await client.dataset(run.defaultDatasetId).listItems();

    console.log(`Fetched ${items.length} tender record(s):`);
    for (const item of items) {
        console.log(
            `- [${item.event_type}] ${item.idCompra} | ${item.reparticion} | ` +
            `${item.tipoCompra} | apertura: ${item.fechaAperturaSobres}`
        );
    }
}

main().catch((err) => {
    console.error('Run failed:', err.message);
    process.exit(1);
});
